### PR TITLE
Support for scram-sha-256 authentication method

### PR DESCRIPTION
**Pull request by zwirek**

### CODE REVIEW

File: tests/Integration/ScramSha256PasswordTest.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+declare(strict_types=1);
+
+namespace PgAsync\Tests\Integration;
+
+use PgAsync\Client;
+use Rx\Observer\CallbackObserver;
+
+class ScramSha256PasswordTest extends TestCase
+{
+    public function testScamSha256Login()
```
Comment (mbonneau) [inline tests/Integration/ScramSha256PasswordTest.php:11]: This method name is missing an 'r'

**Review by mbonneau — COMMENTED** [review]
This looks good. Thanks for taking the time to work on this. I did not have a chance to look at why the test was timing out.

File: src/PgAsync/Message/Message.php [hunk 2]
```diff
@@ -23,11 +23,11 @@ public static function prependLengthInt32(string $s): string
         return Message::int32($len + 4) . $s;
     }
 
-    public static function createMessageFromIdentifier(string $identifier): ParserInterface
+    public static function createMessageFromIdentifier(string $identifier, array $dependencies): ParserInterface
```
[the remark this answers is not in the document]
Comment (zwirek) [reply src/PgAsync/Message/Message.php]: I have taken this as an internal method but you are right. This should have default value

File: tests/Integration/ScramSha256PasswordTest.php (hunk 1 above)
Comment (zwirek) [reply tests/Integration/ScramSha256PasswordTest.php]: This is definitely not a scam :)

**Comment by zwirek** [comment]
Thanks for the review. My overlook CI configuration is causing the failed tests. The only thing I have changed is docker-compose to have an extra service with postgres 15 for local development

In order to tests start to pass, I have changed CI configuration and setup postgres to have additional user with scram-sha-256 authorization method. I have added additional postgres versions as 15 is the latest.

**Comment by Neustradamus** [comment]
@zwirek: Really a good job! Thanks.

And @mbonneau too for reviewing!

**Pull request merged**